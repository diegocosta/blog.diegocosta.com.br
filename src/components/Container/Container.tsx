import React from 'react';
import styled, { createGlobalStyle } from 'styled-components';

import Header from '../Header';
import Footer from '../Footer';

const Wrapper = styled.section`
  margin: auto;
  max-width: 800px;
  padding: 20px;
`;

const GlobalStyle = createGlobalStyle`
  body {
    background: #0e0f11;
    font-family: 'Raleway', sans-serif;
    color: #fff;
  }

  .gist-data { border: 0px !important; }

  .gist-data tbody, .gist .highlight, .gist .gist-data { background-color: #0e0f11 !important; }

  .gist .gist-file { border: 0px !important; }

  .gist-meta { background: transparent !important; }

  .gist-data tbody td:nth-of-type(1) { color: #2B91AF !important; }

  .gist-data tbody td:nth-of-type(2){ color: #FFFFFF !important; }

  .pl-c { color: #57A64A !important; }

  .pl-k, tbody tr:first-child .blob-code, tbody tr:last-child .blob-code { color: #569CD6 !important; }

  .pl-en { color: #FFFFFF !important; }

  .pl-c1 { color: #FFFFFF !important; }

  .pl-pds { color: #D69D85 !important; }

  .pl-s { color: #D69D85 !important; }
`;

export interface ContainerProps {
  children: React.ReactElement;
  small?: boolean;
}

export default function Container(props: ContainerProps): React.ReactElement {
  return (
    <Wrapper>
      <GlobalStyle />
      <Header small={props.small || false} />
      <main>{props.children}</main>
      <Footer />
    </Wrapper>
  );
}
