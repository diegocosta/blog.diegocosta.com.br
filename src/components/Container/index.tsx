import React from 'react';

import Container, { ContainerProps } from './Container';

export default (props: ContainerProps): React.ReactElement => {
  return <Container small={props.small}>{props.children}</Container>;
};
