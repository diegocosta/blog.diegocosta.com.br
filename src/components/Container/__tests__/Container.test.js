import React from 'react';
import { render } from '@testing-library/react';

import Container from '../Container';

jest.mock('../../Header', () => () => <p>Header</p>);
jest.mock('../../Footer', () => () => <p>Footer</p>);
jest.mock('../../Search', () => () => <p>Search</p>);
jest.mock('../../Navigation', () => () => <p>Navigation</p>);

describe('<Container />', () => {
  it('should render properly', () => {
    const { baseElement, queryByText } = render(<Container>John Doe</Container>);
    expect(queryByText('John Doe')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
