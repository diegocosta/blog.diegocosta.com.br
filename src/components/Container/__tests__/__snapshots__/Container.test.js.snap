// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render properly 1`] = `
.c0 {
  margin: auto;
  max-width: 800px;
  padding: 20px;
}

@media (min-width:760px) {
  .c1 {
    display: none;
  }
}

<body>
  <div>
    <section
      class="c0"
    >
      <p>
        Header
      </p>
      <div
        class="c1"
      >
        <p>
          Navigation
        </p>
      </div>
      <p>
        Search
      </p>
      <main>
        John Doe
      </main>
      <p>
        Footer
      </p>
    </section>
  </div>
</body>
`;
